src/sonic-ctrmgrd/ctrmgr/container

```diff
@@ -0,0 +1,367 @@
+    owner, if fallback is enabled and remote_state==none, it starts
+    the local image using docker, which will run until kube
+    deployment occurs.
+
+    :param feature: Name of the feature to start.
+    
+    """
+    ret = 0
+    debug_msg("BEGIN")
+    set_owner, fallback = read_config(feature)
+    _, remote_state, _ = read_state(feature)
+
+    debug_msg("{}: set_owner:{} fallback:{} remote_state:{}".format(
+        feature, set_owner, fallback, remote_state))
+
+    update_data(feature, {SYSTEM_STATE: "up"})
+
+    if set_owner == "local":
+        # Ensure label is not there.
+        set_label(feature, False)
+        if remote_state.lower() != "none":
+            update_data(feature, {REMOTE_STATE: "none"})
+        ret = docker_action("start", feature)
+
+    else:
+        if fallback and (remote_state == "none"):
+            # Start local as fallback enabled & no transition occuring.
+            ret = docker_action("start", feature)
+        set_label(feature, True)
+
+    debug_msg("END")
+    return ret
+
+
+def container_stop(feature):
+    """
+    Stops the running container for this feature.
+
+    Instruct/ensure kube terminates, by removing label, unless
+    an kube upgrade is happening.
+
+    Gets the container ID for this feature and call docker stop.
+
+    Marks the feature state down in STATE-DB FEATURE table.
+
+    :param feature: Name of the feature to stop.
+    
+    """
+    debug_msg("BEGIN")
+    set_owner, _ = read_config(feature)
+    current_owner, remote_state, _ = read_state(feature)
+    docker_id = container_id(feature)
+    remove_label = (remote_state != "pending") or (set_owner == "local")
+
+    debug_msg("{}: set_owner:{} current_owner:{} remote_state:{} docker_id:{}".format(
+        feature, set_owner, current_owner, remote_state, docker_id))
+
+    update_data(feature, {SYSTEM_STATE: "down"})
+
+    if remove_label:
+        set_label(feature, False)
+
+    if docker_id:
+        docker_action("stop", docker_id)
+
+    debug_msg("END")
+
+
+def container_kill(feature):
+    """
+    Kills the running container for this feature.
+
+    Instruct/ensure kube terminates, by removing label.
+
+    Gets the container ID for this feature and call docker kill.
+
+    Marks the feature state down in STATE-DB FEATURE table.
+
+    :param feature: Name of the feature to kill.
+    
+    """
+    debug_msg("BEGIN")
+    set_owner, _ = read_config(feature)
+    current_owner, remote_state, _ = read_state(feature)
+    docker_id = container_id(feature)
+    remove_label = (set_owner != "local") or (current_owner != "local")
+
+    debug_msg("{}: set_owner:{} current_owner:{} remote_state:{} docker_id:{}".format(
+        feature, set_owner, current_owner, remote_state, docker_id))
+
+    update_data(feature, {SYSTEM_STATE: "down"})
+
+    if remove_label:
+        set_label(feature, False)
+
+    if docker_id:
+        docker_action("kill", docker_id)
```

Comment:
what is docker_id is null, better to print warning here?

Reply:
Sure.

Reply:
done